pkg/config/template.go

```diff
@@ -47,3 +70,105 @@ func getTemplates(templatePath string) (CiTemplates, error) {
 func isYAML(file string, info os.FileInfo) bool {
 	return !info.IsDir() && (filepath.Ext(file) == ".yaml" || filepath.Ext(file) == ".yml")
 }
+
+// TemplateCMManager holds the details needed for the configmap controller
+type TemplateCMManager struct {
+	cmclient  corev1.ConfigMapInterface
+	prNumber  int
+	logger    *logrus.Entry
+	templates CiTemplates
+}
+
+// NewTemplateCMManager creates a new TemplateCMManager
+func NewTemplateCMManager(cmclient corev1.ConfigMapInterface, prNumber int, logger *logrus.Entry, templates CiTemplates) *TemplateCMManager {
+	return &TemplateCMManager{
+		cmclient:  cmclient,
+		prNumber:  prNumber,
+		logger:    logger,
+		templates: templates,
+	}
+}
+
+// CreateCMTemplates creates configMaps for all the changed templates.
+func (c *TemplateCMManager) CreateCMTemplates() error {
+	var errors []error
+	for filename, template := range c.templates {
+		templateData, err := GetTemplateData(template)
+		if err != nil {
+			errors = append(errors, err)
+		}
+
+		cmName := GetTempCMName(template.Name, []string{filename, templateData})
```

Comment:
I would prefer the `filename` and `templateData` to be proper arguments instead of the API requiring a slice (why should I care as a caller) but whatever.

Reply:
addressed